src/module/iotjs_module_fs.c

```diff
@@ -243,14 +243,14 @@ JHANDLER_FUNCTION(Read) {
 
   int fd = JHANDLER_GET_ARG(0, number);
   const iotjs_jval_t* jbuffer = JHANDLER_GET_ARG(1, object);
-  int offset = JHANDLER_GET_ARG(2, number);
-  int length = JHANDLER_GET_ARG(3, number);
+  size_t offset = JHANDLER_GET_ARG(2, number);
+  size_t length = JHANDLER_GET_ARG(3, number);
```

Comment:
I think the condition `(offset >= data_length)` may be false. For example, on 32-bit architecture, If a user passes `INT_MIN` (-2147483648) to offset, `INT_MIN` will be interpreted as `-INT_MIN` (+2147483648) after casting to size_t. If `data_length` is larger than` -INT_MIN`, the user will read some value at the unintended large offset (It is want I mentioned as unintended operation on above comment). What about checking the sign of offset and throwing exception on negative integer? I think the `length` has the same problem.